include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -155,6 +155,22 @@ __pattern_transform_scan_base_impl(_ExecutionPolicy&& __exec, _Iterator1 __first
         .wait();
 }
 
+template <typename _Iterator1, typename _Iterator2>
+constexpr bool
+__check_equal_iterators(_Iterator1 it1, _Iterator2 it2)
+{
+    using _Decay_Iterator1 = std::decay_t<_Iterator1>;
+    using _Decay_Iterator2 = std::decay_t<_Iterator2>;
+
+    if constexpr (std::is_same_v<_Decay_Iterator1, _Decay_Iterator2>)
+    {
+        return it1 == it2;
+    }
+
+    // In-place exclusive scan works correctly only if an input and an output iterators are the same type.
```

Comment:
I think the comment describes the situation non quite right. 
Lets add the following text:
"Otherwise, there is no way to check an in-place case and a workaround below is not applied."

Reply:
Fixed,